src/Cache/Engine/FileEngine.php

```diff
@@ -388,7 +388,7 @@ protected function _setKey($key, $createKey = false)
         if (!$createKey && !$path->isFile()) {
             return false;
         }
-        if (empty($this->_File) || $this->_File->getBasename() !== $key) {
+        if (null === $this->_File || $this->_File->getBasename() !== $key) {
```

Comment:
As a side note: In the past it was far more valuable to remove the cloaking empty() and isset() checks in favor of !
As !$this->_File behaves exactly the same, but any typo would be recognized immediately - whereas with the cloaking checks typos could lead to dead never-executable code without you knowing it (we had that issue in the past).